### ISSUE TITLE
What does these codes marked as "Workaround for iOS anamorphic image" used for

### ISSUE THREAD

**Issue by devshen** [rs/SDWebImage#1481]
```
#ifdef TARGET_OS_IPHONE
            // Workaround for iOS anamorphic image
            if (partialImageRef) {
                const size_t partialHeight = CGImageGetHeight(partialImageRef);
                CGColorSpaceRef colorSpace = CGColorSpaceCreateDeviceRGB();
                CGContextRef bmContext = CGBitmapContextCreate(NULL, width, height, 8, width * 4, colorSpace, kCGBitmapByteOrderDefault | kCGImageAlphaPremultipliedFirst);
                CGColorSpaceRelease(colorSpace);
                if (bmContext) {
                    CGContextDrawImage(bmContext, (CGRect){.origin.x = 0.0f, .origin.y = 0.0f, .size.width = width, .size.height = partialHeight}, partialImageRef);
                    CGImageRelease(partialImageRef);
                    partialImageRef = CGBitmapContextCreateImage(bmContext);
                    CGContextRelease(bmContext);
                }
                else {
                    CGImageRelease(partialImageRef);
                    partialImageRef = nil;
                }
            }
#endif
```

what does these codes used for while SD has `decodedImageWithImage` method already

**Comment by bpoplauschi** [comment]
Not sure. This code is from 2012. Maybe @rs can answer to this.

**Comment by rs** [comment]
Was introduced in https://github.com/rs/SDWebImage/commit/ba71333e17e18ecdef26574963150e3bd3591fca from PR #114

Was about handling of progressive download. Maybe it no longer relevant.

**Comment by dreampiggy** [comment]
This code...Seems not make sence. I'd suggest to remove that and use the common `shouldDecodeImages` instead.

**Comment by dreampiggy** [comment]
OK....I found the original blog about this strange code. See https://cocoaintheshell.whine.fr/2011/05/progressive-images-download-imageio/

It seems that on the earily version of Image/IO, the behavior does not match the current one. If the image is not PNG, it will return the actual partial height of total images(Like what I talk above). So if you directlly use that image, you will see the stretch version of partial images depends on your imageView's content mode.

However, In my test, iOS 8+ above Image/IO always return the full height images, with the transparent (Or white if you do not support alpha channel) space filling the unfinished height. Like this one during progressive decoding (lldb debug print):

<img width="1552" alt="2018-04-25 1 32 39" src="https://user-images.githubusercontent.com/6919743/39227779-24a1c3ca-488d-11e8-9e8c-8540d9afbb9f.png">

The original source images is https://s3.amazonaws.com/fast-image-cache/demo-images/FICDDemoImage001.jpg . You can see the both partial image or full image height is the same `766`.

So I think, maybe now we can remove that hack code. Because duplicated force decode may impact performance, and may cause the image distortion. We should avoid force decode same image multiple times.

**Comment by dreampiggy** [comment]
@devshen @bpoplauschi I create the PR about this, see #2303

**Comment by dreampiggy** [comment]
Fix by #2303
